File: Python-For-Everyone-Horstmann/Chapter3-Decisions/P3.12.py
# Write a program that translates a letter grade into a number grade. Letter grades are
# A, B, C, D, and F, possibly followed by + or –. Their numeric values are 4, 3, 2, 1, and
# 0. There is no F+ or F–. A + increases the numeric value by 0.3, a – decreases it by 0.3.
# However, an A+ has value 4.0.
#     Enter a letter grade: B-
#     The numeric value is 2.7.

from sys import exit

grade = str(input("Enter your grade: "))

score = 0

if grade[0] == "A":
    score = 4.0

elif grade[0] == "B":
    score = 3.0

elif grade[0] == "C":
    score = 2.0

elif grade[0] == "D":
    score = 1.0

else:
    exit("Unknown grade. Exiting...")

if len(grade) == 2:
    if grade[1] == "+"  and (grade[0] != "A" and grade[0] != "F"):
        score += 0.3

    elif grade[1] == "-" and grade[0] != "F":
        score -= 0.3

elif len(grade) > 2:
    exit("Invalid grade. Exiting...")

print("Your score is:", score)

File: Java-for-Everyone-Late-Objects-Horstmann/Chapter-3-Decisions/P3_12.java
/*Write a program that translates a letter grade into a number grade. Letter grades are
 A, B, C, D, and F, possibly followed by + or –. Their numeric values are 4, 3, 2, 1, and
 0. There is no F+ or F–. A + increases the numeric value by 0.3, a – decreases it by 0.3.
 However, an A+ has value 4.0.
     Enter a letter grade: B-
     The numeric value is 2.7.*/

import java.util.Scanner;

public class P3_12 {
	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		final double A_GRADE = 4.0;
		final double B_GRADE = 3.0;
		final double C_GRADE = 2.0;
		final double D_GRADE = 1.0;
		final double F_GRADE = 0.0;
		final double PLUS_MINUS_ADJUSTMENT = 0.3;
		System.out.print("Letter grade: ");
		String grade = input.next();
		input.close();
		double numeric_grade = F_GRADE;
		if (grade.charAt(0) == 'A') {
			numeric_grade = A_GRADE;
		} else if (grade.charAt(0) == 'B') {
			numeric_grade = B_GRADE;
		} else if (grade.charAt(0) == 'C') {
			numeric_grade = C_GRADE;
		} else if (grade.charAt(0) == 'D') {
			numeric_grade = D_GRADE;
		} else if (grade.charAt(0) == 'F') {
		} else {
			System.out.println("Invalid letter grade.");
			return;
		}

		if (grade.charAt(1) == '+') {
			numeric_grade += PLUS_MINUS_ADJUSTMENT;
		} else if (grade.charAt(1) == '-') {
			numeric_grade -= PLUS_MINUS_ADJUSTMENT;
		} else {
			System.out.println("Invalid letter grade");
			return;
		}
		System.out.printf("\nLetter grade %s = %f", grade, numeric_grade);
	}
}
